mautrix_instagram/db/message.py

```diff
@@ -62,18 +64,29 @@ async def delete_all(cls, room_id: RoomID) -> None:
     @classmethod
     async def get_by_mxid(cls, mxid: EventID, mx_room: RoomID) -> Message | None:
         q = (
-            "SELECT mxid, mx_room, item_id, client_context, receiver, sender "
+            "SELECT mxid, mx_room, item_id, client_context, receiver, sender, ig_timestamp "
             "FROM message WHERE mxid=$1 AND mx_room=$2"
         )
         row = await cls.db.fetchrow(q, mxid, mx_room)
         if not row:
             return None
         return cls(**row)
 
+    @classmethod
+    async def get_last(cls, mx_room: RoomID) -> Message | None:
+        q = (
+            "SELECT mxid, mx_room, item_id, client_context, receiver, sender, ig_timestamp "
+            "FROM message WHERE mx_room=$1 AND ig_timestamp IS NOT NULL ORDER BY ig_timestamp DESC LIMIT 1"
```

Comment:
This should also check `AND item_id NOT LIKE 'fi.mau.instagram.%'` just to be safe